api/helpers/gamestate/moves/resolve/two.js

```diff
@@ -0,0 +1,54 @@
+module.exports = {
+  friendlyName: 'Resolve Two One-Off',
+
+  description: 'Returns new GameState resulting from resolving a two, which scraps target royal and glasses eight on the field, and counter one-off cards.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'The latest GameState before the two resolves',
+      required: true,
+    },
+  },
+  sync: true, // synchronous helper
+  fn: ({ currentState }, exits) => {
+    let result = _.cloneDeep(currentState);
+
+
+    // Checks if target card is facecard or jack
+    if (result.oneOffTargetType === 'jack') {
+
+      //getting cardIndex from opponents points
+      const cardIndex = result.p1.points.findIndex((card) => {
+        const attachIndex = card.attachments.findIndex((attachment) => {
+          return attachment.id === result.targetCard.id;
+        });
+        return attachIndex !== -1; // Check if the attachment exists in the card
+      });
+      
+      if (cardIndex !== -1) {
+        
+        //remove the card from opponents points
+        const [pointCard] = result.p1.points.splice(cardIndex, 1);
+        
+        //removing the jack from point cards attachment
+        pointCard.attachments = pointCard.attachments.filter((attachment) => {
+          return attachment.id !== result.targetCard.id;
+        });
```

Comment:
The removed jack also needs to be added to the scrap. In order to reduce loops, it might make sense to replace the above .findIndex() calls with regular for loops that set variables for both the pointIndex and jackIndex so that down here, if the point card was found, we can use the jackIndex we already calculated to splice the jack from the point's attachments and then we can add the jack to the scrap

Reply:
in two.js what i did was that after condtion finish for faceCard or jack. it will just take result.targetCard in push it in scrap.  
on line 58. ```result.scrap.push(result.targetCard);```